shiny/render/_data_frame_utils/_tbl_data.py

```diff
@@ -321,7 +349,7 @@ def _(
         else slice(None)
     )
     indx_rows = rows if rows is not None else slice(None)
-    return data[indx_rows, indx_cols]
+    return data[indx_cols][indx_rows]
```

Comment:
we just pushed out a new Narwhals version which supports passing a tuple of sequences to `__getitem__`, so if you upgrade your Narwhals version then `data[indx_rows, indx_cols]` should work